lib/android-helpers.js

```diff
@@ -419,7 +419,19 @@ helpers.pushSettingsApp = async function (adb, throwError = false) {
 
 helpers.pushUnlock = async function (adb) {
   logger.debug("Pushing unlock helper app to device...");
-  await adb.installOrUpgrade(unlockApkPath, UNLOCK_HELPER_PKG_ID, true);
+
+  // the unlock helpers does not have the correct versioning, so
+  // `adb.installOrUpgrade` will not work
+  try {
+    await adb.install(unlockApkPath, false);
```

Comment:
I'd think we can use the same strategy to install both settings and unlock helpers: 
installOrUpgrade first, uninstall and install if this fails and then just warn instead of throwing exception.
These helpers are not so important in general, that we must stop the whole session because of an error there.

Also, I'd be happy if we could customise the error message that is thrown (if you still would like it to be there). It should be more helpful and should suggest users to uninstall the corresponding tool manually from the device in order to solve the problem.

Reply:
Good ideas

Reply:
I'm fine with putting a warning and not throwing an exception, so the session can continue.

As for using the same logic for both, that is not possible. The unlock helper will _never_ be upgraded by the `installOrUpgrade` function, since the version of the app is never updated. Every published version of the `appium-unlock` helper comes out as version 1 when using `aapt`.

Reply:
that's sad. Can we improve this tool to have appropriate versioning?
@vrunoa @vmaxim ping

Reply:
@mykola-mokhnach I get correct `versionCode` from `AndroidManifest.xml` with `aapt`:
```
C:\1\unlock_apk>type AndroidManifest.xml | findStr version*
<?xml version="1.0" encoding="utf-8"?>
      android:versionCode="555"
      android:versionName="1.555">

C:\1\unlock_apk>%android_home%\build-tools\26.0.0\aapt dump badging bin\unlock_apk-debug.apk | findStr version*
package: name='io.appium.unlock' versionCode='555' versionName='1.555' platformBuildVersionName='4.4.2-1456859'```

Reply:
@mykola-mokhnach appium/unlock_apk/pull/8

Reply:
cool, thanks @vmaxim 

Now we need to think about how do we deal here with upgrade from older builds, which don't have the version code set